pyqir/pyqir/_native.pyi

```diff
@@ -986,3 +846,242 @@ def result_type(context: Context) -> Type:
     :returns: The result type.
     """
     ...
+
+# Runtime
+
+def array_record_output(builder: Builder, num_elements: Value, label: Value) -> None:
+    """
+    Inserts a marker in the generated output that indicates the start
+    of an array and how many array elements it has.
+
+    :param Builder builder: The IR Builder used to create the instructions
+    :param Value num_elements: How many array elements the array has
+    :param str label: A string label for the array. Depending on the output schema, the label is included in the output or omitted.
```

Comment:
Should be type Value, not str. But you don't need types in the docstrings since this file has type annotations, so you could remove them here.